src/Protocol/Protocol18x.cpp

```diff
@@ -1075,7 +1075,7 @@ void cProtocol180::SendRespawn(eDimension a_Dimension, bool a_ShouldIgnoreDimens
 	if ((m_LastSentDimension == a_Dimension) && !a_ShouldIgnoreDimensionChecks)
 	{
 		// Must not send a respawn for the world with the same dimension, the client goes cuckoo if we do (unless we are respawning from death)
-		return;
```

Comment:
How about you change this line to `SendRespawn(a_Dimension != eDimension::dimOverworld ? eDimension::dimOverworld : eDimension::dimNether, false);` That should handle #3114.